### ISSUE TITLE
Protocol parameter invalid in typing_extensions==4.6

### ISSUE THREAD

**Issue by giovp** [astropenguin/xarray-dataclasses#201]
with python 3.10 and typing_extensions 4.6 I get this error
```
.tox/py310-linux/lib/python3.10/site-packages/xarray_dataclasses/typing.py:74: in <module>
      class DataClass(Protocol[PInit]):
  /opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/typing.py:312: in inner
      return func(*args, **kwds)
  .tox/py310-linux/lib/python3.10/site-packages/typing_extensions.py:672: in __class_getitem__
      raise TypeError(
  E   TypeError: Parameters to Protocol[...] must all be type variables. Parameter 1 is ~PInit
  ```

referred to this line https://github.com/astropenguin/xarray-dataclasses/blob/d57fa4145c3e3d2a379035f062097325848f72be/xarray_dataclasses/typing.py#L74

**Comment by astropenguin** [comment]
Dear @giovp , thank you for reporting this issue. It seems that it is a bug related to `typing_extension` (https://github.com/python/typing_extensions/issues/181), not to our side. Now they released the bug-fixed version (v4.6.1) and it does not cause the error (at least in my environment). So could you try to update typing_extension`?

**Comment by giovp** [comment]
it works, thanks a lot!
